projects/core/src/cms/store/actions/page.action.ts

```diff
@@ -1,48 +1,66 @@
-import { Action } from '@ngrx/store';
+import { PAGE_DATA_ENTITY } from '../cms-state';
+import {
+  EntitySuccessAction,
+  EntityLoadAction,
+  EntityFailAction
+} from '../../../state';
 import { Page } from '../../model/page.model';
 import { PageContext } from '../../../routing/index';
 
+export const LOAD_PAGE_INDEX = '[Cms] Load Page Index';
+export const LOAD_PAGE_INDEX_FAIL = '[Cms] Load Page Index Fail';
+export const LOAD_PAGE_INDEX_SUCCESS = '[Cms] Load Page Index Success';
 export const LOAD_PAGEDATA = '[Cms] Load PageData';
 export const LOAD_PAGEDATA_FAIL = '[Cms] Load PageData Fail';
 export const LOAD_PAGEDATA_SUCCESS = '[Cms] Load PageData Success';
-export const REFRESH_LATEST_PAGE = '[Cms] Refresh latest page';
-export const UPDATE_LATEST_PAGE_KEY = '[Cms] Update latest page key';
-export const CLEAN_PAGE_STATE = '[Cms] Clean Page State;';
 
-export class LoadPageData implements Action {
-  readonly type = LOAD_PAGEDATA;
-  constructor(public payload: PageContext) {}
+export class LoadPageIndex extends EntityLoadAction {
+  readonly type = LOAD_PAGE_INDEX;
+  constructor(public payload: PageContext) {
+    super(payload.type, payload.id);
+  }
 }
 
-export class LoadPageDataFail implements Action {
-  readonly type = LOAD_PAGEDATA_FAIL;
-  constructor(public payload: any) {}
+export class LoadPageIndexFail extends EntityFailAction {
+  readonly type = LOAD_PAGE_INDEX_FAIL;
+  constructor(pageContext: PageContext, public payload: any) {
+    super(pageContext.type, pageContext.id, payload);
+  }
 }
 
-export class LoadPageDataSuccess implements Action {
-  readonly type = LOAD_PAGEDATA_SUCCESS;
-  constructor(public payload: { key: string; value: Page }) {}
+export class LoadPageIndexSuccess extends EntitySuccessAction {
+  readonly type = LOAD_PAGE_INDEX_SUCCESS;
+  constructor(pageContext: PageContext, public payload: string) {
+    super(pageContext.type, pageContext.id, payload);
+  }
 }
 
-export class RefreshLatestPage implements Action {
-  readonly type = REFRESH_LATEST_PAGE;
+export class LoadPageData extends EntityLoadAction {
```

Comment:
Only the index data has flags, right? Page data does not have it. So, it should just `implements Action`, right?

Reply:
Hmmm, good point. @dunqan ?

Reply:
It should implement `EntityAction` instead

Reply:
In that case  we can remove `entutyType` from these actions, but
what should we use here: `pageData: entityReducer<Page>(PAGE_DATA_ENTITY, fromPageReducer.reducer),`?